tasks/install_powershell.json

```diff
@@ -33,6 +33,11 @@
     "stop_service": {
       "description": "Whether to stop the puppet agent service after install",
       "type": "Optional[Boolean]"
+    },
+    "retry": {
+      "description": "The number of download tries",
```

Comment:
I would go with something more generic and explicit like: `The number of retries in case of network connectivity failures`.
With this in mind, I would also add retry logic in the `install_file` method to every `yum`, `zypper` and `apt-get` commands. Failures are seen there also in the collected data attached to the ticket.
Reminder: add this whole retry parameter to the `install.json` file also.

Reply:
good observation!